main.tf

```diff
@@ -239,7 +239,7 @@ resource "aws_rds_cluster" "secondary" {
 }
 
 resource "aws_rds_cluster_instance" "default" {
-  count                                 = local.cluster_instance_count
+  count                                 = contains(regex("^(?:.*(aurora))?.*$", var.engine), "aurora") ? local.cluster_instance_count : 0
```

Comment:
It may be good to put this regex as an input to this module.

Reply:
done. added the regex as input. i have added the test `examples/postgres`. So in the test the regex is at input level only now.